### DISCUSSION TITLE
Optimizing app.zip with tree shaking?

### DISCUSSION THREAD

**Discussion by hololeo** [flet-dev/flet#3119] [Ideas]
A minimal "hello flet" web when compiled with flet build web produces an app.zip of 3 megs. Inside is the entirety of flet with controls not used by the app.

This produces a long startup time making the user experience feel laggy even broken

Is there a possibility of 'tree shaking' so the resulting app.zip only contains the necessary flet code that is used by the application.  

The idea would be to do a source analysis and only use the stuff that is imported.

Or allow the developer to list out the components they want to include.

**Comment by baseplate-admin** [comment]
I don't think it is possible at this point in time..

For context :

* https://github.com/kivy/kivy-github-discussions/issues/26
* https://github.com/orgs/kivy/discussions/22

Duplicate of :

* https://github.com/flet-dev/flet/issues/3056

**Comment by hololeo** [comment]
intresting. though i dont think its a duplicate of #3056 because this request optimization of the flet build web process. with flet build web --wasm we are getting close to sub 3 second start up, but scrunching down app.zip would be the last mile

**Comment by FeodorFitsner** [comment]
My guts tell me tree-shaking for Python is non-trivial task. Some tools, such as PyInstaller (or maybe Nuitka?) tries to solve that, but it's not 100% possible to analyze all imports.

**Comment by baseplate-admin** [comment]
> intresting. though i dont think its a duplicate of https://github.com/flet-dev/flet/issues/3056 because this request optimization of the flet build web process. with flet build web --wasm we are getting close to sub 3 second start up, but scrunching down app.zip would be the last mile

My apologies, i saw `app.zip`, my brain told me it was related to `serious-python`, and i am pretty much involved with the `android` side of `serious-python`


> My guts tell me tree-shaking for Python is non-trivial task. 

The problem with python is those pesky C extensions, those are pretty much the size gobblers.
